### PR TITLE
Rename TEST-31-OOMPOLICY to avoid conflict and look in $BOOT for kernels

### PR DESCRIPTION
**Pull request by keszybz**
This got broken somehow in 44e6a5ef825.

### CODE REVIEW

File: src/core/manager.c [hunk 1]
```diff
@@ -2524,8 +2524,14 @@ static int manager_dispatch_sigchld(sd_event_source *source, void *userdata) {
                 /* Finally, execute them all. Note that u1, u2 and the array might contain duplicates, but
                  * that's fine, manager_invoke_sigchld_event() will ensure we only invoke the handlers once for
                  * each iteration. */
-                if (u1)
+                if (u1) {
+                        /* We check for oom condition, in case we got SIGCHLD before the oom notification.
+                         * We only do this for the the cgroup the PID belonged to. */
+                        if (u1)
```
Comment (yuwata) [inline src/core/manager.c]: `if (u1)` is redundant.

File: src/core/manager.c [hunk 2]
```diff
@@ -2524,8 +2524,14 @@ static int manager_dispatch_sigchld(sd_event_source *source, void *userdata) {
                 /* Finally, execute them all. Note that u1, u2 and the array might contain duplicates, but
                  * that's fine, manager_invoke_sigchld_event() will ensure we only invoke the handlers once for
                  * each iteration. */
-                if (u1)
+                if (u1) {
+                        /* We check for oom condition, in case we got SIGCHLD before the oom notification.
+                         * We only do this for the the cgroup the PID belonged to. */
```
Comment (yuwata) [inline src/core/manager.c]: the the

**Comment by keszybz** [comment]
Thanks. Updated.

**Comment by yuwata** [comment]
Maybe, split out OOMPOLICY related commits to another PR. Then we can easily review them.

**Comment by keszybz** [comment]
Which commits do you think should be split out?

**Comment by yuwata** [comment]
OK. Not a matter, but I cannot review the last commit. But others seems almost trivial for me. So, if at least the last commit is moved to other PR, then I can easily review bootctl and several test cleanups part.

**Comment by keszybz** [comment]
I removed the last two commits, I'll submit them separately.

File: src/boot/bootctl.c [hunk 3]
```diff
@@ -1067,10 +1067,16 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
 
                 case 'p':
+                        if (arg_print_dollar_boot_path)
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                       "--print-boot-path/-x cannot be combined with --print-esp-path/-p");
```
Comment (yuwata) [inline src/boot/bootctl.c:1072]: Maybe `-p cannot be combined with -x`? Though, of course, it is logically equivalent.

Comment (keszybz) [reply src/boot/bootctl.c:1072]: My thinking was that if I say `bootctl --print-boot-path ...` and I get an error about `-x` I might be confused if I don't know that `-x` is a shortcut for that. I think it's better to be explicit here.

Comment (yuwata) [reply src/boot/bootctl.c:1072]: I mean the place of -x and -p.

Comment (keszybz) [reply src/boot/bootctl.c:1072]: Oh, now I understand. I don't think this matters.

Comment (yuwata) [reply src/boot/bootctl.c:1072]: Yeah.

**Pull request merged**